### DISCUSSION TITLE
Graph databases

### DISCUSSION THREAD

**Discussion by MajidHeydari** [falconry/falcon#1864] [Q&A]
Does not falcon support graphic databases like neo4j?

**Comment by vytas7** [comment]
Hi @MajidHeydari !
No, Falcon does not, itself, ship any ORM inside the framework (in fact, Falcon has no hard dependency on any 3rd party package).
See also our FAQ: [Why doesn’t Falcon come with batteries included?](https://falcon.readthedocs.io/en/latest/user/faq.html#why-doesn-t-falcon-come-with-batteries-included)

Re Neo4j, I'm no expert on the subject, but AFAICT people often use that in conjunction with GraphQL.
Although Falcon primarily promotes the REST architectural style, our media handling is often flexible enough to also afford RPC-like protocols (like GraphQL is).
See also a related discussion on our Gitter channel: https://gitter.im/falconry/dev?at=5fc29944226043667c1f6fc8
As well as this issue: https://github.com/falconry/falcon/issues/690#issuecomment-660443672

**Edit:** given the open-ended nature of your question, I have now converted it to a discussion.
